### DISCUSSION TITLE
Are all themes still not supported because i couldnt find some themes in codium :-)

### DISCUSSION THREAD

**Discussion by Greatz08** [VSCodium/vscodium#1821] [General]
?

**Comment by ghost** [comment]
VSCodium uses open-vsx.org as the default extension market. I mean, you can manually change this to MS market. Just modify the `product.json` as shown in previous issues and discussions.

I'd be pleased if this is of any help :)

**Comment by nilslindemann** [comment]
_Answer by Google AI ([Original](https://share.google/aimode/XpX44GCFLna2gXtN2)) (I tested this on Windows 11, and it works):_

To temporarily switch your VSCodium marketplace to the official Microsoft Visual Studio Marketplace, you must modify or create a `product.json` file in your user configuration directory.

### How to Temporarily Change the Marketplace

1.  **Locate your configuration directory** based on your OS:
    1. **Windows**: `%APPDATA%\VSCodium`
    2. **macOS**: `~/Library/Application Support/VSCodium`
    3. **Linux**: `~/.config/VSCodium`
2.  Create or edit `product.json` in that folder and insert the following code:

    ```json
    {
      "extensionsGallery": {
        "serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
        "cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
        "itemUrl": "https://marketplace.visualstudio.com/items"
      }
    }
    ````

3.  **Restart VSCodium.** You can now search for and install extensions like "Empty Themes" directly from the official marketplace.
4.  **To revert:** Simply delete the `product.json` file or rename it (e.g., `product.json.bak`) and restart VSCodium to return to Open VSX.

### Is it possible to use both side by side?

**No, not natively.** VSCodium's core logic only allows one active extensionsGallery endpoint at a time. However, there are two ways to manage this "side by side":

* **Extension Marketplace Switcher:** You can install the [Extension Marketplace Switcher ](https://open-vsx.org/extension/jacob-ae/vscodium-marketplace-toggle) from Open VSX. This adds a toggle to your status bar or command palette to swap between Open VSX and the Microsoft Marketplace without manual file editing.
* **Manual VSIX Install:** You can keep Open VSX as your default and manually download the .vsix file for "Empty Themes" from the Microsoft Marketplace website. In VSCodium, go to the Extensions view, click the **three dots (...)**, and select **"Install from VSIX..."**.

**Note on Licensing:** Microsoft's Terms of Use technically restrict the use of their marketplace to official "Visual Studio Products". Some proprietary extensions (like Pylance or Live Share) may still fail to run on VSCodium even if installed this way due to hardcoded checks for the official VS Code binary.

**Reply by daiyam** [reply]
I recommend [VSIX Manager](https://open-vsx.org/extension/zokugun/vsix-manager) to install extensions from additional public/private marketplaces, from GitHub or your local files.

**Reply by nilslindemann** [reply]
Yeah, it is used by much more people, and it is more powerful. I found it too complex, as I just want to switch to the MS marketplace sometimes.
